spec/memoize_method_spec.rb

```diff
@@ -112,6 +112,123 @@ def calc
     expect(obj.calc).to be 2
   end
 
+  context 'validates the results according to the validation sample rate' do
+    it 'validates the results according if the percentage is 100' do
```

Comment:
Can we reduce the code duplication in these tests? Maybe factor out the class definition and assertion that the cache validation sampler/out of date handler is called?

Reply:
done -- factor out into different methods